### PR TITLE
Compose services

### PR DESCRIPTION
**Pull request by obeleh**
### What does this PR do?

Ads functionality to manage docker-compose service definitions as proposed in #44911 

### What issues does this PR fix or reference?

#44911

### New Behavior

the following salt calls:

- dockercompose.service_create
- dockercompose. service_upsert
- dockercompose.service_remove
- dockercompose.service_set_tag

Different in the behaviour with all other salt calls in this module is that this has to do a yaml.dumps because we're modifying content and not modifying strings. I've explicitly stated in the docstrings that this is the case:

`This wil re-write your yaml file. Comments will be lost. Indentation is set to 2 spaces`

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by rallytime** [comment]
re-run lint

**Comment by rallytime** [comment]
re-run lint

**Review by terminalmage — CHANGES_REQUESTED** [review]
Now that https://github.com/saltstack/salt/pull/45190 has been merged, we should wait for @rallytime to merge that forward into develop, and then this should be rebased so that we use `salt.utils.yaml.dump()` to do the yaml dumping.

**Review by terminalmage — APPROVED** [review]
I'll work on the yaml dumper, we shouldn't be adding that YAML tag in our custom dumper.

**Comment by terminalmage** [comment]
Oh sorry, I marked this PR as approved but I had a few changes I wanted to see made before merging.

**Review by terminalmage — CHANGES_REQUESTED** [review]
Accidentally marked as approved.  See my earlier comments for a few changes that should be made.

File: salt/modules/dockercompose.py
Comment (terminalmage) [inline salt/modules/dockercompose.py]: Please change this to `safe_dump`.

File: salt/modules/dockercompose.py
Comment (terminalmage) [inline salt/modules/dockercompose.py]: If you use `safe_dump` above, you can replace this with `'{0}:{1}'.format(image, tag)`.

**Review by terminalmage — CHANGES_REQUESTED** [review]
It turns out that using a regular dump here uses PyYAML's default dumper. Using our custom safe dumper, that YAML tag should not be present in the dumped output.

**Comment by rallytime** [comment]
@obeleh Can you come back to this PR and address the comments above?

**Comment by obeleh** [comment]
Sorry for the delay. At the time you asked for the change I was busy with other things. I've tested the change you suggested and it works. Hope everything is fine now :)

**Comment by rallytime** [comment]
re-run lint

**Review by terminalmage — APPROVED** [review]

**Comment by rallytime** [comment]
@obeleh There's a very small lint error here: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/19740/violations/file/salt/modules/dockercompose.py/

Can you fix that?

**Pull request merged**
